RoadMap_SwiftUI/RoadMap_SwiftUI/View/DescriptionView.swift

```diff
@@ -0,0 +1,141 @@
+// DescriptionView.swift
+// Copyright © RoadMap. All rights reserved.
+
+import SwiftUI
+
+/// Экран описания товара
+struct DesriptionView: View {
+    // MARK: - Public property
+
+    var body: some View {
+        VStack {
+            topGradientView
+            furnitureItemImageView
+            selectedItemPriceRectangleView
+                .padding(.top, Constants.Numbers.paddingStandard)
+        }
+        .navigationBarBackButtonHidden()
+        .navigationBarTitleDisplayMode(.inline)
+        .offset(y: isTextEditing ? Constants.Numbers.keyboardOffset : Constants.Numbers.zeroFloat)
+    }
+
+    // MARK: - Private properties
+
+    @FocusState private var isTextEditing
+    @StateObject private var viewModel = DescriptionViewModel()
+
+    private var textEditorView: some View {
+        VStack {
+            ProgressView(
+                "\(Constants.Texts.charsText)\(viewModel.totalCharacters)" +
+                    "\(Constants.Texts.maximumTextEditorCharactersText)",
+                value: Double(viewModel.totalCharacters),
+                total: Double(Constants.Numbers.maximumTextEditorCharacters)
+            )
+            .frame(width: Constants.Numbers.progressViewWidth)
+            .padding()
+            .foregroundColor(.black)
+            .tint(.yellow)
+            TextEditor(text: $viewModel.text)
+                .focused($isTextEditing)
+                .foregroundColor(.black)
+                .font(Font.custom(
+                    Constants.Texts.textEditorCustomFont,
+                    size: Constants.Numbers.descriptionTextEditorFontSize,
+                    relativeTo: .body
+                ))
+                .frame(
+                    width: Constants.Numbers.textEditorWidth,
+                    height: Constants.Numbers.textEditorHeight,
+                    alignment: .center
+                )
+                .onChange(of: viewModel.text) { text in
+                    viewModel.totalCharacters = text.count
+                    viewModel.checkMaximumCharacters()
+                }
+        }
+    }
+
+    private var topGradientView: some View {
+        getTopGradientView()
```

Comment:
```suggestion
private var topGradientView: some View {
        getTopGradientView()
```
лучше так не делать вьюха во вьюги